src/libLookup/Lookup.cpp

```diff
@@ -3172,6 +3172,16 @@ void Lookup::SendTxnPacketToNodes(uint32_t numShards) {
 
           LOG_GENERAL(INFO, "Sent to node " << get<SHARD_NODE_PEER>(*it));
         }
+        if (m_mediator.m_ds->m_shards.at(i).empty()) {
+          continue;
+        }
+        uint16_t lastBlockHash = DataConversion::charArrTo16Bits(
+            m_mediator.m_txBlockChain.GetLastBlock().GetBlockHash().asBytes());
+        uint32_t leader_id =
+            lastBlockHash % m_mediator.m_ds->m_shards.at(i).size();
+        toSend.push_back(get<SHARD_NODE_PEER>(
+            m_mediator.m_ds->m_shards.at(i).at(leader_id)));
+        LOG_GENERAL(INFO, "leader id " << leader_id);
       }
```

Comment:
May be u need to change j < NUM_NODES_TO_SEND_LOOKUP --> j < NUM_NODES_TO_SEND_LOOKUP -1,
otherwise we end up sending to NUM_NODES_TO_SEND_LOOKUP + 1 nodes

Reply:
that is usually a low number like 3 right, anyways